src/rpc/WorkQueue.hpp

```diff
@@ -116,6 +126,11 @@ class WorkQueue {
 
                 func(yield);
                 --curSize_.get();
+                if (curSize_.get().value() == 0 && stopping_) {
+                    auto onTasksComplete = onQueueEmpty_.lock();
```

Comment:
Any chance the callback onQueueEmpty will be call twice?

Reply:
There is no such chance because `curSize_` is atomic and `onQueueEmpty_` is protected by `Mutex`.
The reason to use mutex is because `onQueueEmpty_` is used from multiple threads: from a thread pool processing io_context's tasks and from a thread calling `stop()`.

Reply:
I see, it is possible for `stop()` to be called after line 128 but before 130, so `onQueueEmpty_` will be called twice.